src/WordsUI.java

```diff
@@ -0,0 +1,138 @@
+import java.awt.*;
+import java.awt.event.*;
+import javax.swing.*;
+
+public class WordsUI {
+	JFrame window;
+	JPanel panel;
+	JPanel grid;
+	JPanel buttons;
+	
+	int windowWidth, windowHeight;		// Dimensions of the window in pixels
+	int boardSize; 						// Dimensions (square) of the board in pixels
+	int numCells;						// Number of rows (columns) in the board (must be odd)
+	static final int minCells = 11;
+	static final int maxCells = 31;
+	static final int zoomStep = 2;		// Must be even
+	int xCenterCell, yCenterCell;		// The board coordinates of the cell depicted in the center
+	static final int moveStep = 2;
+	
+	private class Grid extends JPanel {
+		int cellSize;					// Dimensions of a cell in pixels
+		
+		private void DrawCell(Graphics2D g2, int xCenter, int yCenter, int xCell, int yCell) {
+			g2.setPaint(new Color(128, 128, 128));
+			g2.setStroke(new BasicStroke());
+			
+			g2.drawRect(xCenter - cellSize/2, yCenter - cellSize/2, cellSize, cellSize);
+			
+			if (xCell == 0 && yCell == 0)
+				g2.fillRect(xCenter - cellSize/2, yCenter - cellSize/2, cellSize, cellSize);
+		}
+		
+		public void paintComponent(Graphics g) {
+			super.paintComponent(g);
+			
+			Graphics2D g2 = (Graphics2D) g;
+			
+			// Background region
+			int x1,y1,x2,y2;			// Boundaries of the board
+			int cx,cy;					// Center of the board
+			
+			x1 = (windowWidth - boardSize)/2;
+			y1 = 20;
```

Comment:
Padding at the top of the window before the board is shown.  Will push an update with clarification.